### ISSUE TITLE
Viewer mode not persisted

### ISSUE THREAD

**Issue by zoran995** [TerriaJS/terriajs#6337]
Setting viewer mode in the init file will force the viewer mode which will override the value persisted in local storage. This is a change from behaviour implemented in #4947 and I don't see documented in changes.md. 

On first look, it seems that the issue is introduced with plugins in TerriaMap (https://github.com/TerriaJS/TerriaMap/pull/582)
https://github.com/TerriaJS/TerriaMap/blame/75f2489f3bb50096a59caeabe4ce8ca00d02e20b/index.js#L78-L88
this changed the order of initialization and persisted map settings are loaded before init data

**Comment by zoran995** [comment]
Also, map URL parameter don't work anymore 

Steps to reproduce
- Try opening http://ci.terria.io/main/?map=2d
- it opens in 3D, but instead it should be 2d

**Comment by na9da** [comment]
Hi @zoran995 , thanks for the bug report. Sorry that the plugin branch change broke this and a few other things. It also introduces a delay between app load and the shared items appearing on workbench. 

The main intention for the change was that plugins might introduce new catalog items which can appear in the share URL. So the plugin loading should happen before share data initialization. My thinking was I could do it in this sequence `App start -> plugin load -> share data initialization`. But that clearly is not working. I think as a quick fix I'll switch it to `Plugin load -> Share data initialization + App start`.

**Comment by na9da** [comment]
Hi @zoran995, I have a tentative fix in #6342. 

Basically, it changes the plugin initialization sequence as `App start -> beforeRestoreAppState -> restore app state from share data`. The second step is a hook which gets run just before app state is restored where we can load the plugins.

Another issue I came across with this change was, because plugin loading happens before `registerMapNavigations()` (which happens during react render), map buttons added by plugins will appear before standard terria buttons. To avoid this I have made `.order` property optional for `ICompositeBarItem`. That way, plugins can leave the order `undefined` and it will appear last.
